frame/staking/src/pallet/mod.rs

```diff
@@ -1622,20 +1622,31 @@ pub mod pallet {
 		///
 		/// Note that parameter `_validator_count` should be the upper bound of the number of
 		/// validators stored and is only used for weights.
-		#[pallet::weight(T::WeightInfo::force_apply_min_commission(*_validator_count))]
+		#[pallet::weight(T::WeightInfo::force_apply_min_commission(*max_validator_count))]
 		pub fn force_apply_min_commission(
 			origin: OriginFor<T>,
-			_validator_count: u32,
+			max_validator_count: u32,
 		) -> DispatchResult {
 			ensure_root(origin)?;
 			let min_commission = MinCommission::<T>::get();
-			Validators::<T>::translate(|_, mut prefs: ValidatorPrefs| {
+			let mut validators_to_update = Vec::with_capacity(max_validator_count as usize);
+
+			for (index, (validator_id, mut prefs)) in Validators::<T>::iter().enumerate() {
+				if index as u32 >= max_validator_count {
+					log!(trace, "max_validator_count is less than the number of validators");
+					break
+				}
```

Comment:
@shawntabrizi does this suffice for bounding? Here is my rational for not just force applying to single validators: https://github.com/paritytech/substrate/pull/10786#issuecomment-1029196447

Reply:
not a good solution.

We will only be able to iterate over the same first n people. How should we touch the last ones?